### PR TITLE
Miscellaneous fixes

### PR DESCRIPTION
**Pull request by jschueller**
These prevent CIs from passing:
- nbsphinx issue: 5f9dfda it is not possible to work-around invalid links without notebook examples anymore when nbsphinx is disabled
- numpy 1.14 array formatting changed: use legacy 1.13 version, plus avoid bug https://github.com/numpy/numpy/issues/10356

### CODE REVIEW

File: python/src/Sample_doc.i.in [hunk 1]
```diff
@@ -55,9 +55,7 @@ Create a Sample from a (2d) array, list or tuple
 and back
 
 >>> np.array(sample)
-array([[ 1.,  2.],
-       [ 3.,  4.],
-       [ 5.,  6.]])
+array...
```
Comment (dbarbier) [inline python/src/Sample_doc.i.in]: This output looks pretty useless now.

Comment (jschueller) [reply python/src/Sample_doc.i.in]: Agreed, removed

**Comment by dbarbier** [comment]
After reading numpy issue and fix, I am not sure that numpy 1.14.1 will provide the exact same output with np.set_printoptions(legacy='1.13').  Maybe we should wait before merging this PR?

**Comment by jschueller** [comment]
A solution is to rewrite the conversion test without relying on diffs, see last commit.

**Comment by jschueller** [comment]
The formatter approach does not work with np>=114 as the blanks changed:
https://circleci.com/gh/openturns/openturns/1258?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by dbarbier** [comment]
My fix does work, but you have to update `python/test/t_Matrix_numpy.expout`

**Comment by jschueller** [comment]
You're right, maybe I missed something the first time I tested that.

**Comment by regislebrun** [comment]
+1

**Pull request merged**